### ISSUE TITLE
Support imported propTypes

### ISSUE THREAD

**Issue by sapegin** [styleguidist/react-styleguidist#917]
Try [react-docgen-external-proptypes-handler](https://github.com/pasupuletics/learning/blob/master/react-docgen-external-proptypes-handler.js), create a repo, publish to npm, and [add to Styleguidist](https://github.com/styleguidist/react-styleguidist/blob/dfcb73de01d2b3ecf0864a95d358b216dcf65618/scripts/schemas/config.js#L86-L90) by default.

More details about the issue: https://github.com/reactjs/react-docgen/issues/33

Related issues: #548, #792

/cc @pasupuletics

**Comment by selrond** [comment]
@sapegin somebody already published a modified version of this: [siddharthkp/react-docgen-external-proptypes-handler](https://github.com/siddharthkp/react-docgen-external-proptypes-handler)

**Comment by peroperje** [comment]
does it work? If yes, could you put an example of integration into `react-styleguidist`?
I have tried set up `react-styleguidist` with `react-docgen-external-proptypes-handler`  to get something like this :

commonPropTypes.js
```js
import PropTypes from 'prop-types'

export default {
  top: PropTypes.string
}
```
MyComponent.js
```js
import commonPropTypes from './commonPropTypes'
const MyComponent = () => {}

MyComponent.propTypes = {
   ...commonPropTypes
}

export default MyComponent

```
but without success.

**Comment by stepancar** [comment]
@peroperje look at MoneyInput propTypes https://github.com/alfa-laboratory/arui-feather/blob/master/src/money-input/money-input.jsx and to config for react-styleguidist
https://github.com/alfa-laboratory/arui-feather/blob/master/styleguide.config.js#L23
It uses https://github.com/alfa-laboratory/library-utils
which allow you use propTypes from another component.

**Comment by peroperje** [comment]
Thanks, @stepancar. 
I have tried but, unfortunately, it doesn't work for me.
I have instaled [library-utils](https://www.npmjs.com/package/library-utils) and when I put :
```js
    propsParser(filePath) {
        return reactDoc(filePath);
    }
```
in my styleguide.config.js all propTypes for all components disappears from the documentation.

I am not sure what I do wrong.

**Comment by stepancar** [comment]
@peroperje could you provide repo where you use react-styleguidist?
I'll try to help you

**Comment by stepancar** [comment]
@Heymdall, do you know about this problem?

**Comment by Heymdall** [comment]
@stepancar We actually don't use this kind of components in our projects. But I will try to look into it today.

**Comment by siddharthkp** [comment]
@peroperje react-docgen only understands React components (not styled components)

Reference: https://github.com/styled-components/styled-components/issues/945

TL;DR: You'd have to use like this:

```jsx
const H1 = styled.h1`
  font-size: 1.5em;
  text-align: center;
`;

const Headline = props => <H1 {...props} />;

Heading.propTypes = {
  top: PropTypes.string
}
```

**Comment by okonet** [comment]
@siddharthkp we actually do support styled-components in styleguidist with `@component` annotation. See https://github.com/styleguidist/react-styleguidist/pull/657

**Comment by okonet** [comment]
Apparently there is an ongoing work to support this natively in react-docgen: https://github.com/reactjs/react-docgen/pull/352
